tutorial/scripts/LFRic/distributed_memory/3_overlapping_comms/pre_cooked/overlap_specific.py

```diff
@@ -0,0 +1,46 @@
+'''A PSyclone transformation script that transforms a specific
+synchronous halo exchanges into an asynchronous halo exchange and
+moves the halo exchange start part as early as possible in the
+schedule in order to maximise the overlap of communication and
+computation. Also outputs a textual view of the transformed PSyIR
+representing the PSy-layer.
+
+This is a kernel-specific implementation that will only work for
```

Comment:
Does it require annexed dofs to be on or off? When I tried using this script I got a TransformationError?

Reply:
Good point. It assumes they are on. I will update to assume they are off and add some words to the docstring